### DISCUSSION TITLE
Solace PubSub+ Event Broker support

### DISCUSSION THREAD

**Discussion by infoShare** [testcontainers/testcontainers-java#6405] [Ideas]
I was thinking about adding support for PubSub+ Event Broker using testcontainers with authorization and simple topics management.

Information:
https://solace.com/products/event-broker/software/

Docker image:
https://hub.docker.com/r/solace/solace-pubsub-standard

What do you think about that ? 
Executing more complex cases (without multiple steps to configure) with additional configuration is not really straightforward and can be simplified.

**Comment by eddumelendez** [comment]
Thanks for the suggestion. I just took a look at the [docs](https://solace.com/products/event-broker/software/getting-started/) and saw the command below

```bash
docker run -d -p 8080:8080 -p 55554:55555 -p 8008:8008 -p 1883:1883 -p 8000:8000 -p 5672:5672 -p 9000:9000 -p 2222:2222 --shm-size=2g --env username_admin_globalaccesslevel=admin --env username_admin_password=admin --name=solace solace/solace-pubsub-standard
```

So, I don't see any complex setup on this and can be done just using `GenericContainer`. If your suggestion is to have something like `RabbitMQContainer` and its helper methods to execute commands, we, right now, are not in favor of this.  However, we can have an example about using solace pubsub and can be listed [here](https://www.testcontainers.org/examples/)

Let us know if in your experience there is something else besides the command is displayed in the docs and can belong to a testcontainers module.

**Comment by infoShare** [comment]
In most of the cases solace is used with additional SSL configuration and this is not possible to execute without creating additional images with certificates etc. 
Helper methods with executing additional commands are just another simplification - solace created using image in example is just useless without additional config.

But I can try to add example with solace configuration to discuss it further.

**Reply by eddumelendez** [reply]
> But I can try to add example with solace configuration to discuss it further.

Please

**Reply by infoShare** [reply]
@eddumelendez, it looks a little bit complicated.
It would be great to have possibility to execute multiple commands in a single session (to replace configuration script for solace) and some flexible way of waiting based on the command execution result (waitOnCommandResult).
https://github.com/testcontainers/testcontainers-java/pull/6432

**Comment by eddumelendez** [comment]
There is a module already, which will be available in the next release. See https://www.testcontainers.org/modules/solace/
